qiskit/dagcircuit/dagnode.py

```diff
@@ -51,6 +51,17 @@ def op(self):
     def op(self, data):
         self._op = data
 
+    @property
+    def name(self):
+        """Returns the Instruction name corresponding to the op for the node, else _name"""
+        if not self.type or self.type != 'op':
+            return self._name
```

Comment:
We could do this instead, which would remove `_name` from the attributes.
```
    @property
    def name(self):
        """Returns the Instruction name corresponding to the op for the node"""
        if self.type and self.type == 'op':
            return self._op.name

    @name.setter
    def name(self, name):
        if self.type and self.type == 'op':
            self._op.name = name
```
As with `node.condition`, there really is no need for `node.name` since only `op` nodes will have names. There are though probably 100 references to `node.name` and the drawers sometimes use `op.name` and `instruction.name` in place of node.name. So could be done, but major job.

Reply:
I went ahead and did the update above, which removes `name` as an actual attribute and uses `@property` and `setter` only. I also removed the name setting on the swaps.